core/lib/state/src/rocksdb/mod.rs

```diff
@@ -131,23 +152,157 @@ impl RocksdbStorage {
+        let (Some(start_from), true) = (
+            self.enum_migration_start_from(),
+            self.enum_index_migration_chunk_size > 0,
+        ) else {
+            return;
+        };
+
+        let started_at = Instant::now();
+        let progress_percentage = BigDecimal::from(100)
+            * u256_to_big_decimal(h256_to_u256(start_from))
+            / u256_to_big_decimal(U256::MAX);
+        tracing::info!(
+            "RocksDB enum index migration is not finished, current progress {}%, starting from key {start_from:?}",
+            progress_percentage.with_prec(5)
+        );
+
+        let mut write_batch = self.db.new_write_batch();
+        let (keys, values): (Vec<_>, Vec<_>) = self
+            .db
+            .from_iterator_cf(StateKeeperColumnFamily::State, start_from.as_bytes())
+            .take(self.enum_index_migration_chunk_size)
+            .filter_map(|(key, value)| {
+                (key.as_ref() != Self::BLOCK_NUMBER_KEY
+                    && key.as_ref() != Self::ENUM_INDEX_MIGRATION_PROGRESS_KEY
+                    && key.as_ref() != Self::ENUM_INDEX_MIGRATION_FINISHED
+                    && value.len() != 40)
+                    .then(|| (H256::from_slice(&key), H256::from_slice(&value[..32])))
+            })
+            .unzip();
+        let indices = conn
+            .storage_logs_dedup_dal()
+            .enum_indices_for_keys(&keys)
+            .await;
+        for ((key, value), index) in keys.iter().zip(values).zip(indices) {
+            write_batch.put_cf(
+                StateKeeperColumnFamily::State,
+                key.as_bytes(),
+                &Self::serialize_state_value(value, index),
+            );
+        }
+
+        let next_key = keys
+            .last()
+            .and_then(|last_key| h256_to_u256(*last_key).checked_add(U256::one()))
+            .map(u256_to_h256);
+        match next_key {
+            None => {
+                write_batch.put_cf(
+                    StateKeeperColumnFamily::State,
+                    Self::ENUM_INDEX_MIGRATION_FINISHED,
+                    &[1u8; 1],
+                );
+                write_batch.delete_cf(
+                    StateKeeperColumnFamily::State,
+                    Self::ENUM_INDEX_MIGRATION_PROGRESS_KEY,
+                );
+                tracing::info!("RocksDB enum index migration finished");
+            }
+            Some(next_key) => {
+                write_batch.put_cf(
+                    StateKeeperColumnFamily::State,
+                    Self::ENUM_INDEX_MIGRATION_PROGRESS_KEY,
+                    next_key.as_bytes(),
+                );
+            }
+        }
+        self.db
+            .write(write_batch)
+            .expect("failed to save state data into rocksdb");
+        tracing::info!(
+            "RocksDB enum index migration chunk took {:?}, migrated {} keys",
+            started_at.elapsed(),
+            keys.len()
```

Comment:
Do we want to meter these values (e.g., latency as a histogram and migrated keys as a counter)?

Reply:
I think logging them is enough